packages/grid/x-data-grid/src/hooks/features/keyboardNavigation/useGridKeyboardNavigation.ts

```diff
@@ -134,15 +138,27 @@ export const useGridKeyboardNavigation = (
         }
 
         case 'ArrowRight': {
-          if (colIndexBefore < lastColIndex) {
-            goToCell(colIndexBefore + 1, getRowIdFromIndex(rowIndexBefore), 'right');
+          if (direction === 'ltr') {
```

Comment:
For better readability, we can extract `goToLeftCell` and `goToRightCell` functions:
```ts
case 'ArrowRight': {
  if (theme.direction === 'rtl') {
    goToLeftCell();
  } else {
    goToRightCell();
  }
  break;
}
```